File: modules/angular2/src/change_detection/jit_proto_change_detector.ts
import {ListWrapper} from 'angular2/src/facade/collection';

import {ProtoChangeDetector, ChangeDetector, ChangeDetectorDefinition} from './interfaces';
import {ChangeDetectorJITGenerator} from './change_detection_jit_generator';

import {coalesce} from './coalesce';
import {createPropertyRecords, createEventRecords} from './proto_change_detector';

export class JitProtoChangeDetector implements ProtoChangeDetector {
  _factory: Function;

  constructor(private definition: ChangeDetectorDefinition) {
    this._factory = this._createFactory(definition);
  }

  static isSupported(): boolean { return true; }

  instantiate(dispatcher: any): ChangeDetector { return this._factory(dispatcher); }

  _createFactory(definition: ChangeDetectorDefinition) {
    var propertyBindingRecords = createPropertyRecords(definition);
    var eventBindingRecords = createEventRecords(definition);
    return new ChangeDetectorJITGenerator(
               definition.id, definition.strategy, this.definition.bindingRecords, propertyBindingRecords, eventBindingRecords,
               this.definition.directiveRecords, this.definition.generateCheckNoChanges)
        .generate();
  }
}

File: modules/angular2/src/change_detection/jit_proto_change_detector.dart
library change_detection.jit_proto_change_detector;

import 'interfaces.dart' show ChangeDetector, ProtoChangeDetector;

class JitProtoChangeDetector implements ProtoChangeDetector {
  JitProtoChangeDetector(registry, definition) : super();

  static bool isSupported() => false;

  ChangeDetector instantiate(dispatcher) {
    throw "Jit Change Detection not supported in Dart";
  }
}
